### PR TITLE
Add get-by-name to artifact get

### PR DESCRIPTION
**Pull request by jhrozek**
- Add SQL search for artifact versions by artifact name
- Add gRPC call to get artifact by name
- artifact get: Get artifact by name

Fixes: #1762

### CODE REVIEW

**Review by rdimitrov — COMMENTED** [review]
Let's use cobra for the checks about the flags (if they are present or mutually exclusive or where at least one should be present out of a pair). We already do that for some commands so you can use them as examples.

**Comment by jhrozek** [comment]
> Let's use cobra for the checks about the flags (if they are present or mutually exclusive or where at least one should be present out of a pair). We already do that for some commands so you can use them as examples.

Good idea, thank you. I added a new patch into the PR for easier review (they'll all be squashed during push)

**Review by rdimitrov — DISMISSED** [review]

File: proto/minder/v1/minder.proto [hunk 1]
```diff
@@ -78,6 +78,16 @@ service ArtifactService {
             auth_scope: OBJECT_OWNER_PROJECT
         };
     }
+
+    rpc GetArtifactByName (GetArtifactByNameRequest) returns (GetArtifactByNameResponse) {
+        option (google.api.http) = {
+            get: "/api/v1/artifact/{name}"
```
Comment (evankanderson) [inline proto/minder/v1/minder.proto:84]: Do we have to worry about names and IDs colliding?  If so, we could add some additional text to this one to ensure it doesn't conflict.

Comment (rdimitrov) [reply proto/minder/v1/minder.proto:84]: I think the expected format for `--name` is `repoOwner/repoName/artifactName` which should not allow for such conflicts, right?

Comment (jhrozek) [reply proto/minder/v1/minder.proto:84]: yes but I think Evan has a point and I was just being sloppy. Let me change the path

Comment (jhrozek) [reply proto/minder/v1/minder.proto]: @evankanderson @rdimitrov one more patch in this PR, WDYT?

Comment (rdimitrov) [reply proto/minder/v1/minder.proto]: Ah, I was too fast to reply to this one. Got it now 👍

**Review by rdimitrov — APPROVED** [review]

File: proto/minder/v1/minder.proto (hunk 1 above)
Comment (jhrozek) [reply proto/minder/v1/minder.proto]: thans for the ack, I'm going to wait for Evan to see if this is what he meant

Comment (jhrozek) [reply proto/minder/v1/minder.proto]: actually it seems like folks would like to do a release, do I'm going to go YOLO and just merge this.

**Pull request merged**